pynamodb/pagination.py

```diff
@@ -170,15 +169,17 @@ def __init__(
         settings: OperationSettings = OperationSettings.default,
     ) -> None:
         self.page_iter: PageIterator = PageIterator(operation, args, kwargs, rate_limit, settings)
-        self._first_iteration = True
         self._map_fn = map_fn
         self._limit = limit
         self._total_count = 0
+        self._items: List = []
```

Comment:
@ikonst Why is this required. I don't see any calls to `self._items` ahead of https://github.com/pynamodb/PynamoDB/pull/1101/files#diff-05a16c49172d4c74ad60089cd28c1ba68f5da9699bbb94ce1ccee0f9e60d9dfdR182

Reply:
You are right. And for all practical reasons, `self._items[X]` will raise just the same (for an empty list).